src/movies.js

```diff
@@ -1,16 +1,88 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(array){
+    const allDirectors = array.map(movie => { 
+    const container = movie.director
+    
+    return container})
+    return allDirectors
+    } 
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array){
+    const container = array.filter( item =>
+    item.director === "Steven Spielberg" &&
+    item.genre.includes('Drama')
+    )
+    return container.length
+    } 
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(array){
+    if((!array.length) || movies.rate === "") {
+    return 0
+    }
+    const allRatings = array.map(movie => { 
+    const container = movie.rate
+    
+    return container})
+    
+    let total = allRatings.reduce(function(sum, item){
+    return sum + item
+    })
+    return Math.round((total/array.length)*100)/100
+    } 
+    
+    console.log (ratesAverage(movies))
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
+function dramaMoviesRate(array){
+    const drama = array.filter( item =>
+    item.genre.includes('Drama')
+    )
+    if (!drama.length){
+    return 0
+    }
+    
```

Comment:
You should interchange line 45-47 with line 42-44, it makes more sense to check first the length of the array before you try to filter it